core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -706,6 +713,17 @@ public GeyserSession(GeyserImpl geyser, BedrockServerSession bedrockServerSessio
      * Send all necessary packets to load Bedrock into the server
      */
     public void connect() {
+        if (this.dimensionType.minY() < BedrockDimension.OVERWORLD.minY() || this.dimensionType.maxY() > BedrockDimension.OVERWORLD.maxY()) {
+            final int minY = Math.max(this.dimensionType.minY(), -512);
+            final int maxY = Math.min(this.dimensionType.maxY(), 512);
+            this.bedrockOverworldDimension = new BedrockDimension(minY, maxY - minY, true, BedrockDimension.OVERWORLD_ID);
+            this.bedrockDimension = this.bedrockOverworldDimension;
+
+            final DimensionDataPacket dimensionDataPacket = new DimensionDataPacket();
+            dimensionDataPacket.getDefinitions().add(new DimensionDefinition("minecraft:overworld", maxY, minY, 5));
+            upstream.sendPacket(dimensionDataPacket);
+        }
```

Comment:
let's add some debug logging here that the world height was extended, just in case this could cause any issues at a later point